File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * 品牌接口.
 *
 * @author 邓鹏涛
 * @date 2019/2/9 14:48
 */
public interface BrandService {

    /**
     * 查询所有品牌.
     * @return
     */
    List<TbBrand> findAll();

    /**
     *分页查询品牌.
     * @param pageNum 当前页
     * @param pageSize 每页显示数量
     * @return
     */
    PageResult findPage(int pageNum, int pageSize);

    /**
     * 新增品牌.
     * @param brand
     */
    void add(TbBrand brand);

    /**
     * 根据主键查找id.
     * @param id
     * @return
     */
    TbBrand findOne(Long id);

    /**
     * 更新品牌.
     * @param brand
     */
    void update(TbBrand brand);

    /**
     * 删除品牌.
     * @param id
     */
    void delete(long id);

    /**
     * 批量删除品牌.
     * @param ids
     */
    void deleteBatch(Long[] ids);

    /**
     * 分页查询条件品牌.
     * @param brand
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult findPage(TbBrand brand, int pageNum, int pageSize);

    /**
     * 前端下拉框需要数据
     * @return
     */
    List<Map> selectOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
// 品牌服务
app.service("brandService", function ($http) {

    this.findAll = function () {
        return $http.get('../brand/all.do');
    };

    this.findPage = function (page,size) {
        return $http.get('../brand/page.do?page='+ page +'&size=' + size)
    };

    this.add = function (entity) {
        return $http.post('../brand/add.do', entity);
    };

    this.update = function (entity) {
        return $http.post('../brand/update.do', entity);
    };

    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id=' + id);
    };

    this.dele = function (ids) {
        return $http.get('../brand/delete.do?ids=' + ids);
    };

    this.search = function (page, size, searchEntity) {
        return $http.post('../brand/search.do?page=' + page + '&size=' + size,searchEntity);
    };

    // 下拉列表数据
    this.selectOptionList = function () {
        return $http.get('../brand/selectOptionList.do');
    }
});